projects/ngx-avatar/src/lib/avatar.component.ts

```diff
@@ -153,18 +151,34 @@ export class AvatarComponent implements OnChanges, OnDestroy {
    *
    * memberOf AvatarComponent
    */
-  public fetchAvatarSource(event?: any): void {
-    const avatarSource = this.sources[this.currentSource];
-    if (!avatarSource) {
+  public fetchAvatarSource(): void {
+    const previousSource = this.sources[this.currentIndex];
```

Comment:
This would not work, because then `previousSource` is a boolean. But we need a `Source | undefined` in order to pass it to `markSourceAsFailed`

Reply:
After checking if it exists we can access the value with `this.sources[this.currentIndex]`, but that's more of a preference as browsers doesn't seem to throw errors about index out of bounds as of today.

Reply:
A

```ts
    const previousSource = this.sources[this.currentIndex];
    if (previousSource) {
      this.avatarService.markSourceAsFailed(previousSource);
    }
```

B

```ts
    if (this.currentIndex in this.sources) {
      this.avatarService.markSourceAsFailed(this.sources[this.currentIndex]);
    }
```

While A is one line longer, I feel the intent of the code is made much clearer, especially by the introduction of a well named variable `previousSource`. B is shorter, but has a risk of introducing bugs in future maintenance because logic is duplicated between the condition the actual access to the value. And I feel B would need to have a comment explaining what we are trying to do.

Reply:
I always have the concern when it comes to accessing array values, probably because my usage of strong typed languages where most they will produce an exception when you try to access an index that doesn't exist in the collection. So you gotta always check first and I came to find it a best practice. As JS doesn't bother to return `undefined` when an index doesn't exist, I think we can go with the original intend (A) and take profit of this.

Reply:
ok, then nothing to be done here